odoo/addons/base/models/ir_sequence.py

```diff
@@ -154,6 +174,29 @@ def _get_current_sequence(self, sequence_date=None):
     use_date_range = fields.Boolean(string='Use subsequences per date_range')
     date_range_ids = fields.One2many('ir.sequence.date_range', 'sequence_id', string='Subsequences')
 
+    @api.constrains('prefix', 'suffix')
+    def _check_prefix_and_suffix(self):
+        for sequence in self:
+            for field_name in ['prefix', 'suffix']:
+                value = sequence[field_name]
+                if not value:
```

Comment:
```suggestion
                if not value or '%' not in value:
```
- Also continue `%` not in value because no need to check another code without it